cookbook/product/query-products.rst

```diff
@@ -59,16 +59,80 @@ Add sorters:
         // sort by completeness, the locale and scope is expected, if not provided, the default one are used
         ->addSorter('completeness', 'DESC', ['locale' => 'fr_FR', 'scope' => 'mobile']);
 
-Execute the query
+Execute the Query to Get a Cursor
+---------------------------------
+
+It will return a `Akeneo\Component\StorageUtils\CursorInterface` on the products collection, this interface is "storage agnostic".
+
+.. code-block:: php
+
+    $productsCursor = $pqb->execute();
+    foreach ($productsCursor as $product) {
+        // your custom logic
+    }
+
+We strongly advise to use this way to execute queries on products, the next one should be deserved to special cases. For instance, when you want to use capabilities only available on the real storage (ex: MongoDB aggregate).
+
+Execute the Query ("classic" Doctrine mode)
+-------------------------------------------
+
+Once filters and sorters applied, you can still access the internal QueryBuilder to manipulate it before to execute the query.
+
+Then you can use a "classic" Doctrine execute (with custom hydration, etc),
+
+.. code-block:: php
+
+    // can be a `Doctrine\ORM\QueryBuilder` or `Doctrine\ODM\MongoDB\Query\Builder`
+    $queryBuilder = $pqb->getQueryBuilder();
+    $queryBuilder->getQuery()->execute();
+
+Use the Paginator
 -----------------
 
+You can also use a paginator to fetch pages of products from the `Cursor`.
```

Comment:
as the cursor is already paginated we should not advise using this paginator. I really don't see the benefit of using it (as you can define the internal page size of the cursor...